Jenkinsfiles/Jenkinsfile.deploy_ami

```diff
@@ -0,0 +1,156 @@
+def aws_creds = 'cbj-deploy'
+def backend_config = ''
+def terraform_vars = ''
+
+pipeline {
+  agent {
+    node {
+      label ''
+      customWorkspace 'blue-button-deploy-ami'
+    }
+  }
+
+  parameters {
+    string(
+      defaultValue: "*/master",
+      description: 'The branch of the deployment repo to use for the build.',
+      name: 'DEPLOY_BRANCH'
+    )
+    string(
+      defaultValue: "",
+      description: 'The Blue Button AMI id that will be used for the deployment.',
+      name: 'AMI_ID'
+    )
+    string(
+      defaultValue: "m3.medium",
+      description: 'The class/size of the ec2 instance to launch.',
+      name: 'INSTANCE_CLASS'
+    )
+    choice(
+      choices: 'dev\ntest\nimpl\nprod',
+      description: 'The environment to deploy to. Required.',
+      name: 'ENV'
+    )
+  }
+
+  stages {
+    stage('Ensure ENV and AMI_ID') {
+      steps {
+        sh """
+        if [ -z "${params.BRANCH}" ] || [ -z "${params.ENV}" ] || [ -z "${params.AMI_ID}" ] || [ -z "${params.SUBNET_ID}" ]
+        then
+          exit 1
+        fi
+        """
+      }
+    }
+
+    stage('Checkout') {
+      steps {
+        checkout([
+          $class: 'GitSCM',
+          branches: [[
+            name: "${params.DEPLOY_BRANCH}"
+          ]],
+          doGenerateSubmoduleConfigurations: false,
+          extensions: [[
+            $class: 'RelativeTargetDirectory',
+            relativeTargetDir: 'code'
+          ]],
+          userRemoteConfigs: [[
+            url: 'https://github.com/CMSgov/bluebutton-web-deployment.git'
+          ]]
+        ])
+      }
+    }
+
+    stage('Determine terraform config files') {
+      steps {
+        script {
+          if (params.ENV == 'dev') {
+            backend_config = 'bb-backend-dev'
+            terraform_vars = 'bb-tf-dev'
+          }
+          if (params.ENV == 'test') {
+            backend_config = 'bb-backend-test'
+            terraform_vars = 'bb-tf-test'
+          }
+          if (params.ENV == 'impl') {
+            backend_config = 'bb-backend-impl'
+            terraform_vars = 'bb-tf-impl'
+          }
+          if (params.ENV == 'prod') {
+            backend_config = 'bb-backend-prod'
+            terraform_vars = 'bb-tf-prod'
+          }
+        }
+      }
+    }
+
+
+    stage('Sanity check terraform plan') {
+      steps {
+        script {
+          dir('code') {
+            withAwsCli(credentialsId: aws_creds, defaultRegion: 'us-east-1') {
+              withCredentials([
+                file(credentialsId: backend_config, variable: 'bc'),
+                file(credentialsId: terraform_vars, variable: 'tv')
+              ]) {
+                sh """
+                  cd terraform/${params.ENV}
```

Comment:
I know this may be a nit/maybe not the normal Jenkins way, but I don't really like having these scripts baked in here like this.  I worry that having the scripts in here rather as independently runnable/independently testable things would make them harder to maintain (and tie the deployment process to Jenkins in a way that's hard to do independently.

Reply:
Good point. I will attempt to move these sections out to scripts that any dev (or Jenkins) can invoke.